### PR TITLE
make fastcheck -- run linters and tests in 30 seconds

### PR DESCRIPTION
**Pull request by tiran**
Are you also annoyed by the fact that you have waited an hour for CI to finish -- only to find out that your PR has a PEP 8 violation in a file? ``make fastcheck`` to your rescue! It checks pylint, pep8 and ipatests' unit tests in about 30 seconds on both (!) Python 2 and 3:

```
$ time make fastcheck
...
real    0m29,417s
user    1m18,018s
sys     0m2,637s
```

- [ ] add ``IPA_GIT_BRANCH`` to release docs
- [x] update https://www.freeipa.org/page/Testing

## Add make targets for fast linting and testing

Fast linting only needs modified files with pylint and diff with
pycodestyle. It's good enough to detect most code errors very fast. It
typically takes less than 10 seconds. A complete full pylint run uses
all CPU cores for several minutes. PEP 8 violations are typically
reported after 30 minutes to several hours on Travis CI.

Fast lintings uses git diff and git merge-base to find all modified
files in a branch or working tree. There is no easy way to find the
branch source. On Travis the information is provided by Travis. For
local development it's a new variable IPA_GIT_BRANCH in VERSION.m4.

Fast testing execute all unit tests that do not depend on ipalib.api.

In total it takes about 30-40 seconds (!) to execute linting, PEP 8 checks
and unittests for both Python 2 and 3.

## Add marker needs_ipaapi and option to skip tests

The new marker needs_ipaapi is used to mark tests that needs an
initialized API (ipalib.api) or some sort of other API services (running
LDAP server) to work. Some packages use api.Command or api.Backend on
module level. They are not marked but rather skipped entirely.

A new option ``skip-ipaapi`` is added to skip all API based tests. With
the option, only simple unit tests are executed. As of now, freeIPA
contains more than 500 unit tests that can be executed in about 5
seconds.

### CODE REVIEW

File: ipatests/test_ipaserver/test_ipap11helper.py [hunk 1]
```diff
@@ -48,6 +48,8 @@
 
 @pytest.fixture(scope="module")
 def p11(request):
+    if not os.path.isfile(SOFTHSM2_UTIL):
```
Comment (abbra) [inline ipatests/test_ipaserver/test_ipap11helper.py]: please make this change as a separate commit

Comment (tiran) [reply ipatests/test_ipaserver/test_ipap11helper.py]: I dropped the two lines.

**Comment by abbra** [comment]
Ok, thanks. LGTM. Let's wait for the builds.

**Comment by abbra** [comment]
I added information about `fastcheck` to https://www.freeipa.org/page/Testing#Fast_test

**Comment by tiran** [comment]
I correct it. You don't need to supply the branch information.

**Comment by tiran** [comment]
master:

* b98f9b46de40e7b486cad32e5562595522d5e1ea Add marker needs_ipaapi and option to skip tests
* 7fbbf6689e5083b14f457d37b2419733826ef2ce Add make targets for fast linting and testing
